src/actions/bookshelf/utils/bookshelf-utils.js

```diff
@@ -124,3 +126,63 @@ export function appendWhereClause (queryBuilder, fieldName, value, matchStrategy
+// queryBuilder - The queryBuilder instance
+// modelName    - The model name of the main resource
+// -----------------------------------------------------------------------------
+// This logic supports ordering by columns of associations (via dot
+// notation). If the association is not defined on the source model, the orderBy
+// argument is ignored.
+//
+// NOTES:
+// + Only supports a depth of 1 (i.e. <association>.<field>).
+// + NULLS are always returned last in both ASC and DESC orders.
+// -----------------------------------------------------------------------------
+export function appendOrderByClause (joint, queryBuilder, modelName, fieldValue) {
+  const orderBy = buildOrderBy(fieldValue)
+
+  orderBy.forEach(orderOpt => {
+    // Support column from association
+    if (orderOpt.col.includes('.')) {
+      const parts = orderOpt.col.split('.')
+      const assocName = parts[0]
+      const colName = parts[1]
+      const assocModelName = (joint.modelNameOfAssoc[modelName]) ? joint.modelNameOfAssoc[modelName][assocName] : null
+
+      if (assocModelName) {
+        // Obtain model config info to build raw query
+        const mainTableName = joint.model[modelName].prototype.tableName
+        const assocTableName = joint.model[assocModelName].prototype.tableName
+        const mainModelConfig = joint.modelConfig.find(it => it.name === modelName)
+        const assocConfig = mainModelConfig.associations[assocName]
+
+        if (!assocConfig) {
+          // Skip is association name does not exist on the model
+          if (debugOrderBy) console.warn(`[JOINT] The orderBy argument "${assocName}.${colName}" is being ignored because the assoctiation "${assocName}" does not exist for model "${modelName}".`)
+        } else if (assocConfig.type !== 'toOne') {
+          // Skip if association is not "toOne" (i.e. it is a "many" relationsip)
+          if (debugOrderBy) console.warn(`[JOINT] The orderBy argument "${assocName}.${colName}" is being ignored because the assoctiation "${assocName}" is not of type "toOne".`)
+        } else {
+          // Include column from association in select statement and perform join with orderBy clause
+          const assocPathInfo = CoreUtils.parseAssociationPath(assocConfig.path)
+          queryBuilder
+            .leftJoin(assocTableName, `${mainTableName}.${assocPathInfo.sourceField}`, `${assocTableName}.${assocPathInfo.targetField}`)
+            .select(`${mainTableName}.*`, `${assocTableName}.${colName}`)
+            .orderByRaw(`${assocTableName}.${colName} IS NULL, ${assocTableName}.${colName} ${orderOpt.order}`)
+        }
+      } else {
+        if (debugOrderBy) console.warn(`[JOINT] The orderBy argument "${assocName}.${colName}" is being ignored because the assoctiation "${assocName}" does not exist for model "${modelName}"`)
+      }
+
+    // Support column from main model
+    } else {
+      queryBuilder.orderBy(orderOpt.col, orderOpt.order)
+    }
+  })
```

Comment:
Following up previous conversation (Github doesn't allow me to include code suggestions in the old thread).

I am suggesting that the program should stop execution here because this kind of error should be caught and fixed during development.

However, I understand the value of collecting all errors and reporting them at once. In this case, a better approach might be to refactor this code into a pattern where the list of operations is built first and then executed separately.

```suggestion
  const results = buildOrderBy(fieldValue)
    .map(orderOpt => {
      // Support column from main model
      if (!orderOpt.col.includes('.')) {
        return [true, (_queryBuilder) => _queryBuilder.orderBy(orderOpt.col, orderOpt.order)]
      }

      // Support column from association
      const parts = orderOpt.col.split('.')
      const assocName = parts[0]
      const colName = parts[1]
      const assocModelName = (joint.modelNameOfAssoc[modelName]) ? joint.modelNameOfAssoc[modelName][assocName] : null

      if (!assocModelName) {
        return [false, `[JOINT] The orderBy argument "${assocName}.${colName}" is being ignored because the assoctiation "${assocName}" does not exist for model "${modelName}"`] 
      }

      // Obtain model config info to build raw query
      const mainTableName = joint.model[modelName].prototype.tableName
      const assocTableName = joint.model[assocModelName].prototype.tableName
      const mainModelConfig = joint.modelConfig.find(it => it.name === modelName)
      const assocConfig = mainModelConfig.associations[assocName]

      if (!assocConfig) {
        // Skip is association name does not exist on the model
        return [false, `[JOINT] The orderBy argument "${assocName}.${colName}" is being ignored because the assoctiation "${assocName}" does not exist for model "${modelName}".`]
      } else if (assocConfig.type !== 'toOne') {
        // Skip if association is not "toOne" (i.e. it is a "many" relationsip)
        return [false, `[JOINT] The orderBy argument "${assocName}.${colName}" is being ignored because the assoctiation "${assocName}" is not of type "toOne".`]
      } else {
        // Include column from association in select statement and perform join with orderBy clause
        const assocPathInfo = CoreUtils.parseAssociationPath(assocConfig.path)
        return [true, (_queryBuilder) => _queryBuilder
          .leftJoin(assocTableName, `${mainTableName}.${assocPathInfo.sourceField}`, `${assocTableName}.${assocPathInfo.targetField}`)
          .select(`${mainTableName}.*`, `${assocTableName}.${colName}`)
          .orderByRaw(`${assocTableName}.${colName} IS NULL, ${assocTableName}.${colName} ${orderOpt.order}`)
        ]
      }
    })

  const failures = results.filter(it => !it[0])
  if (failures.length > 0) {
    // perhaps worth a custom error class?
    throw new Error('Failed to build orderBy clause:\n\t' + failures.map(it => it[1]).join('\n\t'))
  }

  // actually apply the orderBy clauses
  results.forEach(it => it[1](queryBuilder))
```